docs/source/traits_user_manual/observers.rst

```diff
@@ -0,0 +1,557 @@
+* Via instance method: By calling |HasTraits.observe| instance method to establish (or remove) change
+  notification handlers.
+
+Via class definition
+--------------------
+
+Observers can be defined for every instance of a |HasTraits| subclass by
+applying the |@observe| decorator on an instance method. It has the following
+signature:
+
+.. py:decorator:: observe(expression[, post_init=False, dispatch="same"])
+
+   Mark an instance method of |HasTraits| as the handler being called
+   when the specified trait change occurs.
+
+   :param expression: A description of the traits are being observed.
+   :type expression: str or list of str or |Expression|
+   :param bool post_init: Whether the change handler is added after the
+    initial object state is set.
+   :param str dispatch: A string dedicating how the handler should be run.
+
+Example:
+
+.. literalinclude:: /../../examples/tutorials/doc_examples/examples/observe_decorator.py
+   :start-after: observe_decorator
+
+Notice that a change event is emitted at instance construction time because
+the initial value is different from the default value.
+
+The *post_init* argument can be used to delay adding the change handler
+until after the object state is set.
+
+.. literalinclude:: /../../examples/tutorials/doc_examples/examples/observe_post_init.py
+   :start-at: @observe
+   :lines: 1-3
+
+Via instance method
+-------------------
+
+The |HasTraits.observe| method is useful for adding or removing change handler on a per
+instance basis. The method has the following signature:
+
+.. py:method:: HasTraits.observe(handler, expression[, remove=False, dispatch="same"])
+
+   Add or remove change handler to be fired when one or many traits change.
+
+   :param handler: The callable to be called when a change occurs. It will receive an event object as its single argument and shoul not raise any exceptions.
+   :type handler: callable(object)
+   :param expression: A description of the traits are being observed.
+   :type expression: str or list of str or |Expression|
+   :param bool remove: If true, remove the change handler for the given expression.
+   :param str dispatch: A string dedicating how the handler should be run.
+
+Example:
+
+.. literalinclude:: /../../examples/tutorials/doc_examples/examples/observe_method.py
+   :start-after: observe_method
+
+See :ref:`observer-expression` for further information on the ``expression``
+parameter.
+See :ref:`observe-handler-signatures` for further information on the signature
+of the notification handler.
+
+.. _observer-expression:
+
+The *expression* Parameter
+--------------------------
+The *expression* parameter in |@observe| provides significant flexibility
+in specifying not just attributes directly on the current object, but also
+attributes on nested objects and containers.
+
+There are two approaches for creating an expression:
+
+* Expressions as objects
+* Text using Traits domain specific language
+
+Expressions as objects
```

Comment:
I'd do the language first, then do the expression object second: start with what 90% of people will probably use, and then give them the advanced way.